File: problems/P-268-missing-number/solution1.cpp
/*
 * LeetCode Problem: 268. Missing Number
 * Difficulty: Easy
 * Topics: Array, Hash Table, Bit Manipulation, Math, Sorting, Binary Search
 * Date Solved: 05/07/2025
*/

#include <bits/stdc++.h>
using namespace std;

class Solution1 {
private:
    bool search(vector<int>& arr, int k) {
        for (int i = 0; i < arr.size(); i++)
            if (arr[i] == k)
                return true;
        return false;
    }

public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();

        vector<int> arr;
        for (int i = 0; i <= n; i++)
            arr.push_back(i);

        for (int i = 0; i <= n; i++)
            if (!search(nums, arr[i]))
                return arr[i];

        return -1;
    }
};

File: problems/P-268-missing-number/solution1.java
/*
 * LeetCode Problem: 268. Missing Number
 * Difficulty: Easy
 * Topics: Array, Hash Table, Bit Manipulation, Math, Sorting, Binary Search
 * Date Solved: 05/07/2025
*/

import java.util.HashMap;

public class solution1 {
  public int missingNumber(int[] nums) {
    int n = nums.length;
    HashMap<Integer, Integer> freqMap = new HashMap<>();

    for (int i = 0; i < n; i++)
      freqMap.put(nums[i], freqMap.getOrDefault(nums[i], 0) + 1);

    for (int i = 1; i <= n; i++)
      if (!freqMap.containsKey(i))
        return i;

    return -1;
  }
}